scripts/globals/spells/geo-regen.lua

```diff
@@ -0,0 +1,28 @@
+-----------------------------------------
+-- Spell: Geo-Regen
+-----------------------------------------
+require("scripts/globals/status")
+require("scripts/globals/geo")
+-----------------------------------------
+
+function onMagicCastingCheck(caster, target, spell)
+    if caster:getPet() ~= nil then
+        return tpz.msg.basic.LUOPAN_ALREADY_PLACED
+    elseif not caster:canUseMisc(tpz.zoneMisc.PET) then
+        return tpz.msg.basic.CANT_BE_USED_IN_AREA
+    else
+        return 0
+    end
+end
+
+
+function onSpellCast(caster, target, spell)
+    local geo_skill = caster:getCharSkillLevel(tpz.skill.GEOMANCY)
+    local spellCost = 74
+    local power = (geo_skill / 20) / 10
```

Comment:
It seems this is off a little too, as `geo_skill / 20` does produce the stated 30 HP per tick at 600 power, but the further `/10` seems to reduce it down to 3.

Reply:
All good now, potency should be calculated correctly for all spells.